File: servico-musica-java/java.grpc/src/main/java/servico/java/grpc/controller/MusicaController.java
package servico.java.grpc.controller;

import io.grpc.stub.StreamObserver;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.grpc.server.service.GrpcService;
import servico.java.grpc.entity.Musica;
import servico.java.grpc.proto.MusicaResponse;
import servico.java.grpc.proto.MusicaServiceGrpc;
import servico.java.grpc.proto.TodasMusicasRequest;
import servico.java.grpc.proto.TodasMusicasResponse;
import servico.java.grpc.repository.MusicaRepository;

import java.util.List;

@GrpcService
public class MusicaController extends MusicaServiceGrpc.MusicaServiceImplBase {

    @Autowired
    private MusicaRepository musicaRepository;

    @Override
    public void todasAsMusicas(TodasMusicasRequest request, StreamObserver<TodasMusicasResponse> responseObserver) {

        List<Musica> musicas = musicaRepository.findAll();

        TodasMusicasResponse.Builder response = TodasMusicasResponse.newBuilder();

        for (Musica musica : musicas) {

            response.addMusicas(
                    MusicaResponse.newBuilder()
                            .setId(musica.getId())
                            .setNome(musica.getNome())
                            .setArtista(musica.getArtista())
                            .build()
            );
        }

        responseObserver.onNext(response.build());
        responseObserver.onCompleted();
    }
}

File: servico-musica-kotlin/kotlin.grpc/src/main/kotlin/servico/kotlin/grpc/controller/MusicaController.kt
package servico.kotlin.grpc.controller

import io.grpc.stub.StreamObserver
import org.springframework.grpc.server.service.GrpcService
import servico.kotlin.grpc.entity.Musica
import servico.kotlin.grpc.proto.MusicaResponse
import servico.kotlin.grpc.proto.MusicaServiceGrpc
import servico.kotlin.grpc.proto.TodasMusicasRequest
import servico.kotlin.grpc.proto.TodasMusicasResponse
import servico.kotlin.grpc.repository.MusicaRepository

@GrpcService
class MusicaController(
    private val musicaRepository: MusicaRepository
) : MusicaServiceGrpc.MusicaServiceImplBase() {

    override fun todasAsMusicas(
        request: TodasMusicasRequest?,
        responseObserver: StreamObserver<TodasMusicasResponse?>
    ) {
        val musicas: MutableList<Musica> = musicaRepository.findAll()

        val response: TodasMusicasResponse.Builder =
            TodasMusicasResponse.newBuilder()

        for (musica in musicas) {
            response.addMusicas(
                MusicaResponse.newBuilder()
                    .setId(musica.id)
                    .setNome(musica.nome)
                    .setArtista(musica.artista)
                    .build()
            )
        }

        responseObserver.onNext(response.build())
        responseObserver.onCompleted()
    }
}